artemis-commons/src/main/java/org/apache/activemq/artemis/utils/collections/LinkedList.java

```diff
@@ -29,4 +29,8 @@
    void clear();
 
    int size();
+
+   void clearID();
+
+   void setIDSupplier(IDSupplier<E> idSupplier);
```

Comment:
The ID will be translated into a specific reference on the linked list.. so I need this as part of the LinkedList...

This LInkedList was pretty much created for the QueueImpl implementation... this is to allow removing with the ACK ID that came from the source mirror instead of doing a scan over the queue for the message acknowledgement.

Reply:
Yes but there is no remove method here, this interface currently lets you do nothing except clear and influence setting of these IDs you cant use in any way here. The remove method wasnt added (unlike on the priority interface as I later noticed - it doesnt have the clearID method instead) so it just seems weird.

Reply:
Ok I now see your comment on my later comment that you are adding it..heh. So many comments ;)